src/communication/websocket/session.cpp

```diff
@@ -88,13 +91,40 @@ void Session::DoRead() {
       }));
 }
 
-void Session::OnRead(boost::beast::error_code ec, size_t /*bytes_transferred*/) {
+bool Session::Authenticate() {
+  try {
+    const auto creds = nlohmann::json::parse(boost::beast::buffers_to_string(buffer_.data()));
+    if (auth_->Authenticate(creds.at("username").get<std::string>(), creds.at("password").get<std::string>())) {
+      authenticated_ = true;
+      return true;
+    }
+  } catch (const nlohmann::json::out_of_range &out_of_range) {
+    spdlog::error("Missing username or password: {}!", out_of_range.what());
+  } catch (const nlohmann::json::parse_error &parse_error) {
+    spdlog::error("Parsing error: {}!", parse_error.what());
```

Comment:
```suggestion
    spdlog::error("Cannot parse JSON containing username and password: {}!", parse_error.what());
```

Or something similar that has some information about what we were trying to parse.